msfdb

```diff
@@ -0,0 +1,816 @@
+  run_psql("alter role #{@options[:msftest_db_user]} createdb")
+  run_psql("alter role #{@options[:msf_db_user]} with password '#{msf_pass}'")
+  run_psql("alter role #{@options[:msftest_db_user]} with password '#{msftest_pass}'")
+  run_cmd("createdb -p #{@options[:db_port]} -O #{@options[:msf_db_user]} -h 127.0.0.1 -U #{@options[:msf_db_user]} -E UTF-8 -T template0 #{@options[:msf_db_name]}",
+          "#{msf_pass}\n#{msf_pass}\n")
+  run_cmd("createdb -p #{@options[:db_port]} -O #{@options[:msftest_db_user]} -h 127.0.0.1 -U #{@options[:msftest_db_user]} -E UTF-8 -T template0 #{@options[:msftest_db_name]}",
+          "#{msftest_pass}\n#{msftest_pass}\n")
+
+  write_db_client_auth_config
+  restart_db
+
+  puts 'Creating initial database schema'
+  Dir.chdir(@framework) do
+    run_cmd("bundle exec rake db:migrate")
+  end
+end
+
+def write_db_client_auth_config
+  client_auth_config = "#{@db}/pg_hba.conf"
+  puts "Writing client authentication configuration file #{client_auth_config}"
+  File.open(client_auth_config, 'w') do |f|
+    f.puts "host    \"#{@options[:msf_db_name]}\"      \"#{@options[:msf_db_user]}\"      127.0.0.1/32           md5"
+    f.puts "host    \"#{@options[:msftest_db_name]}\"  \"#{@options[:msftest_db_user]}\"  127.0.0.1/32           md5"
+    f.puts "host    \"postgres\"  \"#{@options[:msftest_db_user]}\"  127.0.0.1/32           md5"
+    f.puts "host    \"template1\"   all                127.0.0.1/32           trust"
+    if Gem.win_platform?
+      f.puts "host    all             all                127.0.0.1/32           trust"
+      f.puts "host    all             all                ::1/128                trust"
+    else
+      f.puts "local   all             all                                       trust"
+    end
+  end
+end
+
+def update_db_port
+  if File.file?(@db_conf)
+    config = YAML.load(File.read(@db_conf))
+    if config["production"] && config["production"]["port"]
+      port = config["production"]["port"]
+      if port != @options[:db_port]
+        puts "Using database port #{port} found in #{@db_conf}"
+        @options[:db_port] = port
+      end
+    end
+  end
+end
+
+def ask_yn(question)
+  loop do
+    print "#{question}: "
+    yn = STDIN.gets
+    case yn
+    when /^[Yy]/
+      return true
+    when /^[Nn]/
+      return false
+    else
+      puts 'Please answer yes or no.'
+    end
+  end
+end
+
+def ask_value(question, default_value)
+  print "#{question}[#{default_value}]: "
+  input = STDIN.gets.strip
+  if input.nil? || input.empty?
+    return default_value
+  else
+    return input
+  end
+end
+
+
+def delete_db
+  if Dir.exist?(@db)
+    puts "Deleting all data at #{@db}"
+    stop_db
+    FileUtils.rm_rf(@db)
+    if File.exist?(@db_conf) && ask_yn("Delete database configuration at #{@db_conf}?")
+      File.delete(@db_conf)
+    end
+  else
+    puts "No data at #{@db}, doing nothing"
+  end
+end
+
+def reinit_db
+  delete_db
```

Comment:
I see that this is "by design", but running `msfdb reinit` deletes the user's data without prompting. This caught me off guard when I got back in and was wondering why my data was not present. I feel like we should prompt the user before deleting even if it is the `reinit` command.

```
jbarnett@nexpose-dev:~/goliath$ ./msfdb reinit
Deleting all data at /home/jbarnett/.msf4/db
Stopping database at /home/jbarnett/.msf4/db
Delete database configuration at /home/jbarnett/.msf4/database.yml?: yes
Creating database at /home/jbarnett/.msf4/db
Starting database at /home/jbarnett/.msf4/db...success
Creating database users
Writing client authentication configuration file /home/jbarnett/.msf4/db/pg_hba.conf
Stopping database at /home/jbarnett/.msf4/db
Starting database at /home/jbarnett/.msf4/db...success
Creating initial database schema
MSF web service is no longer running
```